File: go/p1_4_diag_squares.go
package main

import . "g2d"

func main() {
	n := ParseInt(Prompt("N? "))
	InitCanvas(Size{400, 400})
	i := 0
	for i < n {
		c := i * 255 / (n - 1)
		p := i * 300 / (n - 1)
		DrawRect(Color{c, 0, 0}, Rect{p, p, 100, 100})
		i += 1
	}
}

File: exercises/home/p1_4_diag_squares.py
#!/usr/bin/env python3
'''
@author Michele Tomaiuolo - http://www.ce.unipr.it/people/tomamic
@license This software is free - http://www.gnu.org/licenses/gpl.html
'''

import g2d
from random import randrange

W, H = 240, 320
SIDE = 100

n = int(g2d.prompt("n? "))  # like input
dx, dc = 0, 0
if n > 1:
    dx = (W - SIDE) / (n - 1)
    dy = (H - SIDE) / (n - 1)
    dc = 255.0 / (n - 1)

g2d.init_canvas((W, H))

i = 0
while i < n:
    color = (i * dc, 0, 0)
    rect = (i * dx, i * dy, SIDE, SIDE)
    g2d.draw_rect(color, rect)
    i += 1

g2d.main_loop()
